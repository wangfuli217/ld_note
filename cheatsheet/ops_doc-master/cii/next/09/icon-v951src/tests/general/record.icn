#SRC: JCON

record simple(f)
record rec(f1, f2)

procedure main()
   local a, b

   a := rec()
   a.f1 := 1
   a.f2 := 2
   write(a.f1, " ", a.f2)
   a := rec(3)
   a.f2 := 4
   write(a.f1, " ", a.f2)
   a := rec(5,6)
   write(a.f1, " ", a.f2)
   a.f1 := 7
   a.f2 := 8
   write(a.f1, " ", a.f2)
   a := rec(9,10,11)
   write(a.f1, " ", a.f2)
   a := rec(11, 12)
   every write(!a)
   every !a := 13
   write(a.f2)

   b := simple(14)
   write(?b)
   ?b := 15
   write(!b)

   b := rec(3, 7)
   every write(b[1 to 3])
   every write(b["f" || (1 to 3)])

   a := rec(1, 2)
   b := rec(3, 4)
   a.f1 +:= 10
   a.f2 +:= 20
   every !b +:= 70
   every writes(" ", !a | !b | "\n")
end
