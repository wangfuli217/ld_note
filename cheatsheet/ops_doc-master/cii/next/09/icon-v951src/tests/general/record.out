1 2
3 4
5 6
7 8
9 10
11
12
13
14
15
3
7
3
7
 11 22 73 74 
